pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -605,12 +610,22 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       List<ProcessingException> exceptions = new ArrayList<>();
       if (numUnavailableSegments > 0) {
         String errorMessage;
+        String routingPolicy;
+        if (offlineRoutingPolicy != null && realtimeRoutingPolicy != null) {
+          routingPolicy = String.format("%s [realtime], %s [offline]", realtimeRoutingPolicy, offlineRoutingPolicy);
+        } else if (offlineRoutingPolicy != null) {
+          routingPolicy = offlineRoutingPolicy;
+        } else {
+          routingPolicy = realtimeRoutingPolicy;
+        }
```

Comment:
It would be cleaner to pull this out into a small helper function.

Reply:
ack